pkg/subctl/cmd/root.go

```diff
@@ -234,3 +240,27 @@ var nodeLabelBackoff wait.Backoff = wait.Backoff{
 	Factor:   1.2,
 	Jitter:   1,
 }
+
+func checkVersionMismatch(cmd *cobra.Command, args []string) error {
+	config, err := getRestConfig(kubeConfig, kubeContext)
+	exitOnError("The provided kubeconfig is invalid", err)
+
+	clientSet, err := submarinerclientset.NewForConfig(config)
+	exitOnError("Unable to get the Submariner config", err)
+
+	submariner, err := clientSet.SubmarinerV1alpha1().Submariners(OperatorNamespace).Get(submarinercr.SubmarinerName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	if submariner != nil && submariner.Spec.Version != "" {
+		subctlVer, _ := goversion.NewVersion(version.Version)
+		submarinerVer, _ := goversion.NewVersion(submariner.Spec.Version)
+
+		if subctlVer != nil && submarinerVer != nil && subctlVer.LessThan(submarinerVer) {
```

Comment:
Should we only fail when the subctl version is less than the Submariner version? I thought we required them to be the same.

Reply:
The CLI should allow backward compatibility. We could define at most 3 minor versions but for now we don't have such policy.

Reply:
No, then it's impossible to do an upgrade for example. 

But that's a good point.
We should probably test over a limited version skew, and only support that.

For example

subctl version N is expected to work with submariner version N-1, and we test that
for commands (show, etc...).

We at least do it for the join/deploy broker commands on the upgrade job. That job is probably a good opportunity to test the show command on that combination.

Reply:
> We could define at most 3 minor versions but for now we don't have such policy.

Okay makes sense, we can do that in a future PR.